src/Microsoft.Framework.Runtime.Hosting/Json/JsonBuffer.cs

```diff
@@ -0,0 +1,301 @@
+            }
+            else if (first == (int)'n')
+            {
+                ReadLiteral(JsonConstants.ValueNull);
+                result.Type = JsonTokenType.Null;
+            }
+            else if (char.IsDigit((char)first) || first == (int)'-')
+            {
+                result.Type = JsonTokenType.Number;
+                result.Value = ReadNumber(first);
+            }
+            else
+            {
+                throw new JsonDeserializerException(string.Format("Illegal character {0:X4}.", first), result.GetPosition());
+            }
+
+            return result;
+        }
+
+        private int ReadNextChar()
+        {
+            while (true)
+            {
+                var value = _reader.Read();
+                _column += 1;
+
+                if (value == -1)
+                {
+                    // This is the end of file
+                    return -1;
+                }
+                else if (value == JsonConstants.LF)
+                {
+                    // This is a new line. Let the next loop read the first charactor of the following line.
+                    // Set position ahead of next line
+                    _column = 0;
+                    _line += 1;
+
+                    continue;
+                }
+                else if (value == JsonConstants.CR)
+                {
+                    // Skip the carriage return.
+                    // Let the next loop read the following char
+                }
+                else
+                {
+                    // Returns the normal value
+                    return value;
+                }
+            }
+        }
+
+        private string ReadNumber(int firstRead)
+        {
+            var buf = new List<char> { (char)firstRead };
+            while (true)
+            {
+                var next = _reader.Peek();
+
+                if (char.IsDigit((char)next) ||
+                    next == JsonConstants.FullStop ||
+                    next == JsonConstants.UpperE ||
+                    next == JsonConstants.LowerE)
+                {
+                    buf.Add((char)ReadNextChar());
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return new string(buf.ToArray());
+        }
+
+        private void ReadLiteral(string literal)
+        {
+            for (int i = 1; i < literal.Length; ++i)
+            {
+                var next = _reader.Peek();
+                if (next != (int)literal[i])
+                {
+                    throw new JsonDeserializerException(
+                        JsonDeserializerResource.Format_UnrecognizedLiteral(literal),
+                        _line, _column);
+                }
+                else
+                {
+                    ReadNextChar();
+                }
+            }
+
+            var tail = _reader.Peek();
+            if (tail != JsonConstants.RightCurlyBracket &&
```

Comment:
Also, if you decide to take such an approach, at least make sure that you do the same for numbers and strings